torchqdynamics/utils/states.py

```diff
@@ -92,7 +92,9 @@ def coherent(dim: int, alpha: complex | Tensor, *, dtype=None, device=None) -> T
                 [0.003+0.j],
                 [0.000+0.j]], dtype=torch.complex128)
     """
-    return displace(dim, alpha, dtype=dtype, device=device) @ fock(dim, 0)
+    return displace(dim, alpha, dtype=dtype, device=device) @ fock(
```

Comment:
I'm guessing that implementing direclty the series expansion of a coherent state would be faster here. Is there a reason why we choose to apply the displacement operator to vacuum (other than being convenient)? 
If not maybe it's  an easy improvement that we could do sometimes

Reply:
From QuTiP's documentation:

> Select method 'operator' (default) or 'analytic'. With the 'operator' method, the coherent state is generated by displacing the vacuum state using the displacement operator defined in the truncated Hilbert space of size 'N'. This method guarantees that the resulting state is normalized. With 'analytic' method the coherent state is generated using the analytical formula for the coherent state coefficients in the Fock basis. This method does not guarantee that the state is normalized if truncated to a small number of Fock states, but would in that case give more accurate coefficients.

So there isn't a method that is always better than the other, but we could give the choice like qutip does.